src/evaluations/evaluations.service.ts

```diff
@@ -1,11 +1,56 @@
 import {Injectable} from '@nestjs/common';
 import {InjectModel} from '@nestjs/sequelize';
 import {Evaluation} from './evaluation.model';
+import {EvaluationDto} from './dto/evaluation.dto';
+import {CreateEvaluationDto} from './dto/create-evaluation.dto';
+import {UpdateEvaluationDto} from './dto/update-evaluation.dto';
 
 @Injectable()
 export class EvaluationsService {
   constructor(
     @InjectModel(Evaluation)
     private evaluationModel: typeof Evaluation
   ) {}
+
+
+async findAll(): Promise<EvaluationDto[]> {
+    const evaluations = await this.evaluationModel.findAll<Evaluation>();
+    return evaluations.map(evaluation => new EvaluationDto(evaluation));
+  }
+
+  async create(createEvaluationDto: CreateEvaluationDto) {
+      const evaluation = new Evaluation();
+      evaluation.version = createEvaluationDto.version;
+      const evaluationData = await evaluation.save();
+      return new EvaluationDto(evaluationData);
+}
+
+  async update(id: number, updateEvaluationDto: UpdateEvaluationDto): Promise<EvaluationDto> {
+    if (this.evaluationModel.findByPk<Evaluation>(id)){
+      return ;
+    }
+    const evaluation = await this.evaluationModel.findByPk<Evaluation>(id);
+    evaluation.version = updateEvaluationDto.version || evaluation.version; //look into sequelize.update function
+    const evaluationData = await evaluation.save();
+    return new EvaluationDto(evaluationData);
+  }
+
+  async remove(id: number): Promise<EvaluationDto> {
+    if (this.evaluationModel.findByPk<Evaluation>(id)){
```

Comment:
If the evaluation exists, it will halt execution of this function and not delete the eval.

There are examples of an exists function in the users service...